progressReport/parser.py

```diff
@@ -138,9 +141,96 @@ def nodes_to_json(node):
+            "class levels": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "name": {"type": "string"},
+                        "color": {"type": "string", "pattern": "^#([A-Fa-f0-9]{6})$"}
+                    },
+                    "required": ["name", "color"]
+                }
+            },
+            "student levels": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "name": {"type": "string"},
+                        "color": {"type": "string", "pattern": "^#([A-Fa-f0-9]{6})$"}
+                    },
+                    "required": ["name", "color"]
+                }
+            },
+            "count": {"type": "integer"},
+            "nodes": {
+                "type": "object",
+                "properties": {
+                    "id": {"type": "integer"},
+                    "name": {"type": "string"},
+                    "parent": {"type": ["string", "null"]},
+                    "children": {
+                        "type": "array",
+                        "items": {
+                            "$ref": "#/definitions/node"
+                        }
+                    },
+                    "data": {
+                        "type": "object",
+                        "properties": {
+                            "week": {"type": "integer"}
+                        },
+                        "required": ["week"]
+                    }
+                },
+                "required": ["id", "name", "parent", "children", "data"]
+            }
+        },
+        "required": ["name", "term", "start date", "class levels", "student levels", "count", "nodes"],
+        "definitions": {
+            "node": {
+                "type": "object",
+                "properties": {
+                    "id": {"type": "integer"},
+                    "name": {"type": "string"},
+                    "parent": {"type": ["string", "null"]},
+                    "children": {
+                        "type": "array",
+                        "items": {
+                            "$ref": "#/definitions/node"
+                        }
+                    },
+                    "data": {
+                        "type": "object",
+                        "properties": {
+                            "week": {"type": "integer"}
+                        },
+                        "required": ["week"]
+                    }
+                },
+                "required": ["id", "name", "parent", "children", "data"]
+            }
+        }
+    }
+    try:
+        validate(instance=data, schema=schema)
+    except ValidationError as e:
+        # Not sure how we should handle the error, so just raising it for now.
+        raise e
```

Comment:
Not sure how we should handle the error, so just raising it for now.

Reply:
I would not try catch it for now, just let it be thrown so that we can handle it as necessary during later invocations

Reply:
Noted!